s2/s2_fuzz_test.go

```diff
@@ -0,0 +1,34 @@
+package s2
+
+import (
+	"github.com/klauspost/compress/internal/fuzz"
+	"testing"
+)
+
+func FuzzS2(f *testing.F) {
+	fuzz.AddFromZip(f, "testdata/fuzz/block-corpus-raw.zip", fuzz.TypeRaw, false)
+	f.Fuzz(func(t *testing.T, data []byte) {
+		concat, err := ConcatBlocks(nil, data, []byte{0})
+		if err != nil || concat == nil {
+			return
+		}
+
+		if len(data) < 9 { //to avoid errors when converting to uint64
```

Comment:
What is this working around? Smells like it is working around a bug.

Reply:
I want to discuss this one before making changes to this file. During the local testing, I observed an error in "EstimateBlockSize" function. When providing input as "\x01\x00\x00\x00\x00\x00\x00\x00", the error occurs in load64 function from encode_all.go file when calling Uint64 method. It occurs in golang standard libraries and crashes the fuzzer.
I may propose to leave this workaround as it is or remove "EstimateBlockSize" function from the fuzz test.

Reply:
That seems like a real problem. I will send a fix.

Reply:
Fix sent in https://github.com/klauspost/compress/pull/867

When tests pass I will merge - you can then marge master and remove the check 👍🏼

Reply:
Merged. You should be able to remove the check now.

Reply:
Merged.